NGitLab.Mock/Clients/PackageClient.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NGitLab.Models;
+
+namespace NGitLab.Mock.Clients
+{
+    internal sealed class PackageClient : ClientBase, IPackageClient
+    {
+        public PackageClient(ClientContext context)
+            : base(context)
+        {
+        }
+
+        public Task<Package> PublishAsync(int projectId, PackagePublish packagePublish, CancellationToken cancellationToken = default)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public IEnumerable<PackageSearchResult> Get(int projectId, PackageQuery packageQuery)
```

Comment:
@meziantou I couldn't see any prior art for returning an `IEnumerable<T>` in any of the existing clients.  Happy to make a change here, just wanted to make sure that we are on the same page before making any changes.

Reply:
Other implementations use `GitLabCollectionResponse<T>`. This supports sync and async enumeration!